src/dpi.rs

```diff
@@ -240,32 +240,89 @@ impl LogicalSize {
     }
 }
 
-impl From<(f64, f64)> for LogicalSize {
+/// Represents all types into/from which LogicalSize can be converted
```

Comment:
Why precisely is `FromPrimitive` and `ToPrimitive` not satisfactory?

Reply:
Because I would either need to 1. downcast at run-time or like I did, add a method which provides an equivalent getter function for all typoes even if some of them need to be handled differently(which is then done in the getter).
Or do you have a better idea to set this up?

Reply:
Yeah, so why isn't num's `ToPrimitive` work? It lets you convert `f64`s to whatever you want, see https://docs.rs/num-traits/0.2.5/src/num_traits/cast.rs.html#323-348 and https://docs.rs/num/0.2.0/num/cast/trait.ToPrimitive.html

Reply:
Does it round the float on its own like the previous did?

Reply:
:woman_shrugging:

Reply:
E.g. this here: https://github.com/rust-windowing/winit/blob/master/src/dpi.rs#L146

Reply:
As I wasnt sure if ToPrimitive rounds it on their own.

Reply:
Ive just changed it - I hope its better now.

Reply:
Afaik, some reading of the `num` source code reveals that `ToPrimitive` rounds down.

Reply:
What do you suggest, how to do it? I guess rounding the size down is probably not the best.

Reply:
I'm honestly not sure. I'd rather not add a trait, but I'm also not sure `ToPrimitive` will work.

Reply:
Is there a test unit for the into/from conversion? Otherwise we could create one based on the previous functionality and see if it works the same as previously.

Reply:
To my knowledge there is no unit test, but adding one should be trivial.